polaris-react/src/components/TopBar/components/Menu/Menu.tsx

```diff
@@ -93,15 +120,141 @@ export function Menu(props: MenuProps) {
       preferredAlignment="right"
     >
       <div className={styles.MenuItems}>
+        {shouldRenderMenuItems ? (
+          <>
+            {titleMarkup}
+            <div className={styles.TopSection}>
+              {indentedSectionMarkup}
+              {otherStoresMarkup}
+            </div>
+          </>
+        ) : null}
+
         <Box width={customWidth}>
           <ActionList
             actionRole="menuitem"
             onActionAnyItem={onClose}
-            sections={actions}
+            sections={shouldRenderMenuItems ? remainingSections : actions}
           />
           {messageMarkup}
         </Box>
       </div>
     </Popover>
   );
 }
+
+function generateMenuItem(
+  {
+    content,
+    id,
+    accessibilityLabel,
+    url,
+    onAction,
+    icon,
+    prefix,
+    suffix,
+    external,
+    truncate,
+    role,
+    active,
+  }: ActionListItemDescriptor,
+  index: number,
+) {
+  const itemMarkup = (
+    <MenuItem
+      key={`${content}-${index}`}
+      content={content}
+      onAction={onAction}
+      id={id}
+      accessibilityLabel={accessibilityLabel}
+      url={url}
+      icon={icon}
+      prefix={prefix}
+      suffix={suffix}
+      external={external}
+      truncate={truncate}
+      role={role}
+      active={active}
+    />
+  );
+
+  if (id === OTHER_STORES_ID) {
+    return {
+      itemMarkup,
+      isOtherStore: true,
+    };
+  }
+
+  return {itemMarkup, isOtherStore: false};
+}
+
+function buildSections(actions: ActionListProps['sections'], indent: boolean) {
+  if (!actions) {
+    return {indentedSectionMarkup: null, remainingSections: []};
+  }
+
+  let otherStoresMarkup: React.ReactNode = null;
+
+  const [storeListSection, ...remainingSections] = actions;
+  const hasStoreListSection = storeListSection?.id === STORE_LIST_SECTION_ID;
+
+  const storeListSectionItemMarkup = hasStoreListSection
+    ? storeListSection?.items.map((item, index) => {
+        const {itemMarkup, isOtherStore} = generateMenuItem(item, index);
+        if (isOtherStore) {
+          otherStoresMarkup = (
+            <Box width="100%" paddingInlineEnd="2" insetInlineEnd="2">
+              {itemMarkup}
+            </Box>
+          );
+          return null;
+        }
```

Comment:
Can you clarify why "other store" need unique handling? I feel like this is too specific to the admin.

Reply:
It was supposed to be the "Switch organization" button before the refactor.